src/public/client.js

```diff
@@ -78,37 +121,43 @@
     var optionButtons = Array.prototype.slice.apply(optionListUl.getElementsByTagName('button'));
 
     optionButtons.forEach(function (button) {
-      var dataIndex = button.getAttribute('data-option-index');
-      if (parseInt(dataIndex, 10) === parseInt(selectedOption, 10)) button.classList.add('selected');
+      var state = button.getAttribute('data-option-index');
+      if (parseInt(state, 10) == controller.selectedOption ) {
+        button.classList.add('selected');
+      }
       else button.classList.remove('selected');
     });
-
-    controller.selectedOption = controller.game.options[selectedOption];
   }
 
   function selectOption(event) {
     var optionIndex = event.target.getAttribute('data-option-index');
-    selectedOption = optionIndex;
+    controller.selectedOption = parseInt(optionIndex);
     updateGameOptionsSelected();
   }
 
   function updateGameOptions(options) {
+
+    if(options.length <= 0) {
+      throw new Error('Invalid option list for controll.')
+    }
+
     wipeElementsFrom(optionListUl);
-    options.forEach(function (option, index) {
+    if(options.indexOf(controller.selectedOption) == -1) {
```

Comment:
`===` not `==`

Reply:
I thought the difference was due to implicit transtyping and i don't know anything that could be implicitly transtyped to `-1`.
Did I missed something ?